### PR TITLE
[ci-visibility] Fix CI Visibility Spec Tests

### PR DESCRIPTION
**Pull request by juan-fernandez**
## Description
* Always pass `BUILDKITE_TAG` in buildkite test cases, even when empty, since that's what buildkite does. Logic doesn't need to change here because empty values are falsy.
* Fix github pipeline url calculation. 
* Use `GITHUB_SERVER_URL` instead of hardcoding github.com.
* Add new user provided git test cases, where the tag value should overwrite the branch one.

## Checklist
- [ ] Added to the correct milestone.
- [x] Tests provided or description of manual testing performed is included in the code or PR.
- [ ] Library documentation is updated.
- [ ] [Corp site](https://github.com/DataDog/documentation/) documentation is updated (link to the PR).

### CODE REVIEW

**Review by brettlangdon — DISMISSED** [review]

File: releasenotes/notes/update-ci-metadata-extraction-7f5259a18a60fa83.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+fixes:
+  - |
+    Only for CI Visibility (``pytest`` integration): Fix calculation of pipeline URL for GitHub Actions.
```
Comment (P403n1x87) [inline releasenotes/notes/update-ci-metadata-extraction-7f5259a18a60fa83.yaml:4]: Missing new line at EOF 🙁

Comment (juan-fernandez) [reply releasenotes/notes/update-ci-metadata-extraction-7f5259a18a60fa83.yaml:4]: is this enforced by `black`?

Comment (P403n1x87) [reply releasenotes/notes/update-ci-metadata-extraction-7f5259a18a60fa83.yaml]: No, black does not check yaml sources; it's more for keeping the git history as clean as possible 🙂

**Review by brettlangdon — APPROVED** [review]

**Comment by juan-fernandez** [comment]
does it look good otherwise @P403n1x87 ?

**Review by P403n1x87 — APPROVED** [review]
👍

**Pull request merged**